Applications/WebPortal/WebApp/library/source/components/modal.tsx

```diff
@@ -0,0 +1,157 @@
+import * as React from 'react';
+import { DisplaySize } from '../display_size';
+
+interface Properties {
+
+  /** The size of the viewport. */
+  displaySize: DisplaySize;
+
+  /** The height of the modal. */
+  height: string | number;
+
+  /** The width of the modal. */
+  width: string;
+
+  /** Called when the modal should be closed. */
+  onClose?: () => void;
+}
+
+/** This is a component that wraps a child component to style it as a modal. */
+export class Modal extends React.Component<Properties> {
+  public static readonly defaultProps = {
+    onClose: () => {}
+  };
+
+  constructor(props: Properties) {
+    super(props);
+    this.outOfBoundsClick = this.outOfBoundsClick.bind(this);
+  }
+
+  public render(): JSX.Element {
+    const modalStyle = (() => {
+      if(this.props.displaySize === DisplaySize.SMALL) {
+        return {...Modal.STYLE.modalSmall};
+      } else {
+        return {...Modal.STYLE.modalLarge,
+          ...{width: this.props.width, height: this.props.height}};
```

Comment:
Remove the ```...```, instead do:

    return {...Modal.STYLE.modalLarge, width: whatever, height: whatever};